### ISSUE TITLE
ExecutionTest::HelperLaneTest failing after HLSL 2021 got enabled on NVidia, AMD and Qualcomm GPUs

### ISSUE THREAD

**Issue by hekota** [microsoft/DirectXShaderCompiler#5667]
**Description**
`ExecutionTest::HelperLaneTest` is failing on AMD, Nvidia and Qualcomm GPUs after HLSL 2021 language version was enabled by default in commit https://github.com/microsoft/DirectXShaderCompiler/commit/634c2f349. Does not repro on Intel (SM 6.5) or Microsoft Basic Render Driver. Since this shows up on multiple GPUs it points to a compiler or test issue rather than a driver issue.

**Steps to Reproduce**

`hcttest exec-filter *HelperLaneTest -verbose -adapter NVidia*`

**Actual Behavior**
```
StartGroup: ExecutionTest::HelperLaneTest
Verifying IsHelperLane in shader model 6.0
Using Adapter:NVIDIA GeForce GTX 1660 Ti
Error: Verify: AreEqual(pTestResults[0].is_helper_00, 0) - Values (1, 0) [File: D:\dxc2\tools\clang\unittests\HLSLExec\ExecutionTest.cpp, Function: ExecutionTest::HelperLaneTest, Line: 10939]
EndGroup: ExecutionTest::HelperLaneTest [Failed]
```

**Expected Behavior**
```
StartGroup: ExecutionTest::HelperLaneTest
Verifying IsHelperLane in shader model 6.0
Using Adapter:NVIDIA GeForce GTX 1660 Ti
Verifying IsHelperLane in shader model 6.6
Using Adapter:NVIDIA GeForce GTX 1660 Ti
EndGroup: ExecutionTest::HelperLaneTest [Passed]
```

**Environment**
- DXC version 634c2f349 or later (main branch from July 28 forward)
- Windows

**Note**

`ExecutionTest::HelperLaneTestWave` is also failing starting with the HLSL 2021 commit, though only on AMD GPU.

**Comment by hekota** [comment]
Shader:
```
        struct HelperLaneTestResult{
            int is_helper_00;
            int is_helper_10;
            int is_helper_01;
            int is_helper_11;
        };
        
        RWStructuredBuffer<HelperLaneTestResult> g_testResults : register(u0);
        
        int ReadAcrossX_DD(int value, bool isLeft) {
          int d = ddx_fine(value);
          return isLeft ? value + d : value - d;
        }
        
        int ReadAcrossY_DD(int value, bool isTop) {
          int d = ddy_fine(value);
          return isTop ? value + d : value - d;
        }
        
        int ReadAcrossDiagonal_DD(int value, bool isLeft, bool isTop) {
          return ReadAcrossY_DD(ReadAcrossX_DD(value, isLeft), isTop);
        }

        struct PSInput {
            float4 pos : SV_POSITION;
        };
        
        PSInput VSMain(float3 pos : POSITION) {
            PSInput r;
            r.pos = float4(pos, 1); 
            return r;
        }

        uint4 PSMain(PSInput input) : SV_TARGET {
          bool isLeft = (input.pos.x < 1.0f);
          bool isTop = (input.pos.y < 1.0f);
          
          for (int i = 0; i < 2; i++) {
            int is_helper = IsHelperLane();
            int is_helper_accross_X = ReadAcrossX_DD(is_helper, isLeft);
            int is_helper_accross_Y = ReadAcrossY_DD(is_helper, isTop);
            int is_helper_accross_Diag = ReadAcrossDiagonal_DD(is_helper, isLeft, isTop);
          
            if (!isLeft && !isTop) { //bottom right pixel writes results
              g_testResults[i].is_helper_00 = is_helper_accross_Diag;
              g_testResults[i].is_helper_10 = is_helper_accross_Y;
              g_testResults[i].is_helper_01 = is_helper_accross_X;
              g_testResults[i].is_helper_11 = is_helper;
            }

            if (i == 0 && isLeft && isTop) // discard top left pixel
              discard;
          }
          return uint4(0xFFFFFFFF, 0xFFFFFFFF, 0xFFFFFFFF, 1);
        }
```

**Comment by hekota** [comment]
Fixed by #5745. Separate issue #5744 filed to track the underlying issue.
